lib/sidekiq_alive/worker.rb

```diff
@@ -1,11 +1,17 @@
+# frozen_string_literal: true
+
 module SidekiqAlive
   class Worker
     include Sidekiq::Worker
     sidekiq_options retry: false
 
     def perform(_hostname = SidekiqAlive.hostname)
+      # Checks if custom liveness probe passes should fail or return false
+      return unless config.custom_liveness_probe.call
+
+      # Writes the liveness in Redis
       write_living_probe
-      # schedule next living probe
+      # schedules next living probe
       self.class.perform_in(config.time_to_live / 2, current_hostname)
```

Comment:
It should probably still do that even if there's a failure. This will allow to "recover" from errors. Otherwise one failed `custom_liveness_probe` will never try.

Reply:
Hi @arathunku,
The failure should be handled by the `custom_liveness_probe` block.
If the intention is to keep de pod running even if some part of the infrastructure is not running should be handled by the returned value in the block.
ex:
```ruby
proc do
  begin
    db_running?
  rescue => DBKindOfOK
    true
  end
end
```

Either way, I think you should not write a custom liveness probe if an error in the block does not restart de pod.
What do you think? is there something I did not consider?

Reply:
:thinking: Maybe I miss something on my end :see_no_evil: 

I'm using something like this:
```
SidekiqAlive.setup do |config|
  config.time_to_live = 60

  config.custom_liveness_probe = proc do
    HealthProbe.call.success?
  end
end
```
In my `HealthProbe`, there're various checks. In k8s cluster, I expect this probe to failure in the beginning when pod is starting and I've got special threshold for that and then during pod's lifetime, they're lowered. I expect SidekiqAlive to continue

Reply:
@arturictus, sorry to be a burden, is there any chance you could take a look at the comment above when you have a moment?

Reply:
Sorry for the delay.
The previous code should work fine. 
The worker will not store the key in redis and your kubernetes livenessProbe will try to restart  the pod.